### DISCUSSION TITLE
Seperate driver packages

### DISCUSSION THREAD

**Discussion by hrueger** [node-dmx/dmx#117] [Ideas]
What do you think about publishing some drivers to npm as a seperate package?
Maybe not everyone wants to add the serialport or socketio dependency to their project...
Especially since node-gyp isn't easy to get working on some devices.

I thought about creating those packages:
@dmx/core
@dmx/enttec-driver
...

We could include the drivers which don't add extra dependencies in the core package or release them separately. 
Or we could bundle the drivers by their dependency:
@dmx/serial-drivers
...

What dl you think?

**Comment by wiedi** [comment]
I agree that these dependencies are unfortunately a bit painful.

What I worry about is discoverability. If one ends up having to install a bunch of separate packages just to get basic functionality that sounds bad as well.
And it would be a breaking change unless we are very careful with shuffling things around and having the "dmx" package still depend on everything.

**Reply by hrueger** [reply]
Yeah, I agree with you 👍
